R/deltaccd_plot.R

```diff
@@ -1,21 +1,33 @@
-calcCorrSimple = function(df, method) {
-  geneNames = setdiff(colnames(df), 'group')
-  df1 = data.frame(stats::cor(as.matrix(df[,geneNames]), method = method),
-                   gene1 = geneNames, stringsAsFactors = FALSE,
-                   check.names = FALSE) %>%
-    tidyr::gather(-gene1, key = gene2, value = rho) %>%
-    dplyr::filter(gene1!=gene2)}
-
-
-calcCorr = function(ematNow, groupVec, method) {
-  df = data.frame(t(ematNow), group = groupVec, stringsAsFactors = FALSE,
-                  check.names = FALSE) %>%
-    dplyr::group_by(group) %>%
-    dplyr::do(calcCorrSimple(., method)) %>%
-    dplyr::ungroup() %>%
-    dplyr::mutate(gene1 = factor(gene1, rownames(ematNow)),
-                  gene2 = factor(gene2, rev(rownames(ematNow))))}
-
+calcCorrSimple = function(dt, method = 'spearman') {
+  
+  geneNames = setdiff(colnames(dt), 'group')
+  
+  dt1 = data.table::data.table(stats::cor(as.matrix(dt[, geneNames])
+                                          , method = method),
+    gene1 = geneNames)
+  dt1 = data.table::melt(dt1, id.vars = 'gene1', measure.vars = 'gene2'
+    , value.name = 'rho')
+  dt1 = dt1[gene1 != gene2]
+  
+  return(dt1)}
+
+
+calcCorr = function(ematNow, groupVec, method = 'spearman') {
+  
+  dt = data.table::data.table(t(ematNow), group = groupVec)
+  
+  dtFinal = foreach::foreach(group = groupVec, .combine = rbind) %do% {
+    
+    dtTmp = calcCorrSimple(dt[group == group], method = method)
+  
+    return(dtTmp)}
+
+  dtFinal = dtFinal[
```

Comment:
I think two separate lines, each using `:=` in the conventional way, would be more readable.